Common/Inc/GlobalFlag.hpp

```diff
@@ -0,0 +1,39 @@
+/*
+ * GlobalFlag.hpp
+ *
+ * Global Flag Header
+ *
+ * Created on: Nov 29, 2022
+ * Author(s): Gordon Fountain
+ */
+
+#ifndef ZPSW3_GLOBAL_FLAG_HPP
+#define ZPSW3_GLOBAL_FLAG_HPP
+
+#include "FreeRTOS.h"
+
+class GlobalFlag {
+    public:
+        GlobalFlag(bool flag_value = false, int timeout_ms = 1) :: TIMEOUT(timeout_ms / portTICK_PERIOD_MS) {
+            flag = flag_value;
+            flagResourceMux = xSemaphoreCreateMutex();
+            readerCountMux = xSemaphoreCreateMutex();
+            serviceQueue = xSemaphoreCreateCounting(10, 1);
```

Comment:
Maybe set initial count to 0 so we can have 10 spots in the line instead of 9? Because nice round number...you know.